init.lua

```diff
@@ -799,6 +820,11 @@ function Form:close_hud(player)
     hud_fs.close_hud(player, self)
 end
 
+function Form:unset_as_inventory_for(player)
+    open_inv_formspecs[player:get_player_name()] = nil
+    player:set_inventory_formspec("") -- TODO should this restore what it was before? This disables inventory as-is until set by something else.
```

Comment:
I'm not sure what you'd like this to do here.

Reply:
Here's a test of what that would look like:

```lua
function Form:unset_as_inventory_for(player)
    local name = player:get_player_name()
    local form_info = open_inv_formspecs[name]
    if form_info and form_info.self == self then
        open_inv_formspecs[name] = nil
        player:set_inventory_formspec("")
    end
end
```

My concern is that it feels like since it's more or less the same as `Form:close`. Perhaps it'd be better to merge them? Or the duplication doesn't matter. Idk.

Reply:
The duplication here is okay, I think that trying to remove it would just make things more complicated.

Reply:
resolved in 0789c66